src/FoodBox.js

```diff
@@ -0,0 +1,55 @@
+import React, { Component } from 'react';
+
+class FoodBox extends Component {
+  state = {
+    quantity: 1,
+  }
+
+  addQuantity = (e) => {
+    const newQuantity = parseInt(e.target.value);
+    this.setState({
+      quantity: newQuantity,
+    })
+  }
+
+  render() {
+    return (
+      <div className="box">
+        <article className="media">
+          <div className="media-left">
+            <figure className="image is-64x64">
+              <img src={this.props.image} alt="food display" />
+            </figure>
+          </div>
+          <div className="media-content">
+            <div className="content">
+              <p>
+                <strong>{this.props.name}</strong> <br />
+                <small>{this.props.calories}</small>
+              </p>
+            </div>
+          </div>
+          <div className="media-right">
+            <div className="field has-addons">
+              <div className="control">
+                <input
+                  className="input"
+                  type="number"
+                  value={this.state.quantity}
+                  onChange={(e) => { this.addQuantity(e) }}
```

Comment:
No necesitas hacer `onChange={(e) => { this.addQuantity(e) }}` para pasar el evento a tu función, basta con hacer esto: `onChange={this.addQuantity}`, el ya mete como parámetro el evento.